controllers/utils/annotationutils/annotations.go

```diff
@@ -0,0 +1,36 @@
+package annotationutils
```

Comment:
>duplicate 'utils' in complete package name, not nice

The new pkg name is annotationutils it's a separate package from utils (i.e triggering a method in that pkg is `annotationutils.MethodName... ` ) so as far as any dependencies  it can have a different name or folder location.
The only connection is that "annotationutils" folder is nested under "utils" folder (IIUC that is what you meant by "complete pkg name").
I just thought it makes sense to nest all util packages under utils folder, but I don't mind changing either package name or nested folder, let me know what do you think will be better alternative than current.

>And why did this move at all

Here:
>- Split annotation utils into a new package, since it's to be used by the webhook and needed to avoid cyclic dependencies

Reply:
let me clarify: why `utils/annotationutils` vs just `utils/annotations` ? `utils` is duplicated

unclear to me how this can resolve a cyclic dependency, but I will have a closer look out of curiousity...

Reply:
>unclear to me how this can resolve a cyclic dependency, but I will have a closer look out of curiousity...

Original cyclic dependency: 
 `"github.com/medik8s/node-healthcheck-operator/controllers/utils"` **-->**  
 `"github.com/medik8s/node-healthcheck-operator/api/v1alpha1"` **-->**   (this one has nothing to do with the annotations functionality but happens from another file that shared the same package) 
 `"github.com/medik8s/node-healthcheck-operator/controllers/utils"` (that last one was added by my code because I wanted webhook to use utils and caused creation of the cyclic dependency)
 
To solve this I've extracted the util part I've needed into a new package `annotationutils`.
Even though the folder of this package is nested under the utils folder (which contains `utils` package)   this is de-facto a new package which has no dependencies to the utils package, so I can safely use it from  `"github.com/medik8s/node-healthcheck-operator/api/v1alpha1"`

Reply:
> Even though the folder of this package is nested under the utils folder (which contains utils package) this is de-facto a new package which has no dependencies to the utils package

This is interesting, thanks!
Too bad that the webhook code is in the API package, that's the root cause for the problem IMHO. But nothing to think about in this PR of course.